src/components/ArticlePage.jsx

```diff
@@ -26,6 +33,20 @@ const ArticlePage = () => {
       });
   }, [articleId]);
 
+  const handleVote = (voteChange) => {
+    if (!isLoggedIn) {
+      return;
+    } else {
+      setVotes(votes + voteChange);
+      setError("");
+
+      voteOnArticle(articleId, voteChange).catch((err) => {
+        setVotes(votes - voteChange);
+        setError("Failed to update vote. Please try again.");
+      });
+    }
```

Comment:
This logic is great and works well. But I couldn't see the voting icon until I logged in! As a developer, I knew to look at your code to find how to see it. But as a regular user, I would not know that I have to log in to vote! I would mention this somewhere on your page to enhance user experience

Reply:
Thank you for your feedback Rayhaan! I will implement your suggestions. :blush: :raised_hands: